### DISCUSSION TITLE
benefits of defining cron jobs per location in autorestic config vs running the command via cron directly?

### DISCUSSION THREAD

**Discussion by arpitgupta** [cupcakearmy/autorestic#415] [General]
I am trying to understand what the pro/con's are of using the os level cron to schedule all the backup's needed vs defining the cron configs in auto restic configs.

I understand the former case will require multiple entries in the crontab on my system.  Is it just a personal choice or are there benefits of defining the cron in autorestic configs that I am missing?

**Comment by dotboris** [comment]
In theory, everything that you can do with location level crons can be done through your system cron with perhaps one exception.

Autorestic stores the state of when it last ran location based crons in its lock file. This information persists through reboots and downtime. So if your server / computer is down while your backups are meant to run, they won't run again when the system boots back up. When using location based crons, autorestic will run the backup the next time `autorestic cron` is run.

I don't think that you can achieve this will the old fashion cron with a crontab. But you can achieve this using systemd's persistent timers. So there's a benefit there.

My guess is that using the location based cron is meant to be more user friendly. People can edit their autorestic config file without having to mess with the crontab which might be scary or difficult for some people while the yaml file is a bit more accessible.
